pennylane_lightning/src/Gates.cpp

```diff
@@ -42,6 +44,28 @@ Pennylane::AbstractGate::AbstractGate(int numQubits)
     , length(exp2(numQubits))
 {}
 
+void Pennylane::AbstractGate::applyKernel(CplxType* state, vector<size_t>& indices, vector<CplxType>& v) {
+    const vector<CplxType>& matrix = asMatrix();
```

Comment:
Possibly a stupid question - now `asMatrix()` is called multiple times in the `for` loop rather than once outside the `for` loop beforehand. However, this shouldn't be a slowdown because we are just re-defining a pointer to an already existing matrix?

Reply:
Yep, that's correct; its just returning a reference to an already instantiated vector, so this should be a trivial operation.